torchinductor/codegen/triton.py

```diff
@@ -919,7 +930,7 @@ def codegen_kernel(self, name=None):
             if tree.prefix != "r" or self.inside_reduction:
                 argdefs.append(f"{tree.prefix.upper()}BLOCK : tl.constexpr")
 
-        code.writeline(f"def {name or 'kernel'}({', '.join(argdefs)}):")
+        code.writeline(f"def {name or '{kernel_name}'}({', '.join(argdefs)}):")
```

Comment:
The reason I named this just `kernel` is so that the same kernel generated by different graphs (or different models entirely) would share the same cache on disk. 
 
Perhaps we could hash the source code to make naming deterministic?

Reply:
Yeah that would work too. I wouldn't expect too many hits from different models or different graphs though

Reply:
@jansel Oh is that why we're generating a bunch of code strings for different kernels now?

Reply:
Bunch of code strings is for caching autotuning results per kernel

Reply:
Kernel numbers still make navigating profiles easier (like, we easily know how many kernels are generated, seeing that kernel10 is called many times is easier than comparing 6-digit hashes etc)

Reply:
Yes, every kernel gets its own file based on the hash of its source code.  So if two models generate the same kernel, it will be shared.

A lot of models have repeating patterns, especially when they come from the same benchmark suite they might literally have different instances of the same `nn.Module` as layers inside them.

I haven't tested how many hits we are getting, though hopefully dynamic shapes will lead to more hits here.

Reply:
> Kernel numbers still make navigating profiles easier (like, we easily know how many kernels are generated, seeing that kernel10 is called many times is easier than comparing 6-digit hashes etc)

Perhaps we could put this behind a debug config?  I worry it could hurt compile times. 

Is there a way to label a kernel after it is generated?

Reply:
I could check the number of kernels generated in torchbench with and without kernel numbers, and yes, we could put this behind debug config. Even in this PR we are labeling a kernel after its generated, it's generated just with `{kernel_name}` in the string, we could replace it either by `kernel` or by `kernelXX`.

Reply:
Perhpas

Reply:
HuggingFace might have more kernel sharing. We have stuff like: `DebertaForMaskedLM`, `DebertaForQuestionAnswering`, `DebertaV2ForMaskedLM`, `DebertaV2ForQuestionAnswering`

There are also cases where we graph break in a loop and have multiple graphs from the same code.

Reply:
Changed to be controlled by config.